### PR TITLE
move mapping api out of preview namespace

### PR DESCRIPTION
**Pull request by RichardIrons-neo4j**
Move all the mapping classes out of the preview namespace and into `Neo4j.Driver.Mapping`.

### CODE REVIEW

File: Neo4j.Driver/Neo4j.Driver/Public/Mapping/MappingFailedException.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 using System;
 
-namespace Neo4j.Driver.Preview.Mapping;
+namespace Neo4j.Driver.Mapping;
 
 public class MappingFailedException : Neo4jException
```
Comment (thelonelyvulpes) [inline Neo4j.Driver/Neo4j.Driver/Public/Mapping/MappingFailedException.cs:20]: we are missing docs on this public class

File: Neo4j.Driver/Neo4j.Driver/Public/Mapping/IMappingBuilder.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@
 using System;
 using System.Linq.Expressions;
 
-namespace Neo4j.Driver.Preview.Mapping;
```
Comment (thelonelyvulpes) [inline Neo4j.Driver/Neo4j.Driver/Public/Mapping/IMappingBuilder.cs:19]: Neo4j.Driver.Mapping.IMappingBuilder<TObject>.Map<TProperty> is missing documentation for `optional` parameter

**Review by thelonelyvulpes — APPROVED** [review]
We just need to add a few docs for release

**Review by AndyHeap-NeoTech — APPROVED** [review]
The find replace appears to have functioned as advertised.

**Comment by darcome** [comment]
Hello, where can I find docs about mapping? Thanks in advance!

**Comment by RichardIrons-neo4j** [comment]
@darcome [Here is a link](https://neo4j.com/docs/api/dotnet-driver/current/api/Neo4j.Driver.Mapping.html) to the API documentation for mapping, which is correct and up to date. We do not have full documentation for the mapping functionality yet, however you can look through some of the pull requests which contain some information. Note that this documentation is now out of date in some regards.

First addition of mapping: https://github.com/neo4j/neo4j-dotnet-driver/pull/729

Addition of constructor mapping: https://github.com/neo4j/neo4j-dotnet-driver/pull/751

And then these changes made for 5.20 are that the mapping is now case sensitive in matching C# symbols to database fields and properties, and that mapping will fail if a property is not found in the record. I hope this information gets you started; feel free to start a discussion if you need help with it.

**Comment by darcome** [comment]
Thank you!

**Pull request merged**